src/matrix/room/BaseRoom.js

```diff
@@ -501,7 +501,9 @@ export class BaseRoom extends EventEmitter {
                 },
                 clock: this._platform.clock,
                 logger: this._platform.logger,
-                powerLevelsObservable: await this.observePowerLevels()
+                powerLevelsObservable: await this.observePowerLevels(),
+                fetchEventFromStorage: eventId => this._readEventById(eventId),
```

Comment:
why inject this? the homeserver already has access to storage ...

Reply:
`_readEventById()` is already defined in BaseRoom (and is used in `BaseRoom.observeEvent()`?
We wouldn't want to duplicate the same logic in Timeline?

Reply:
Ahh, yes, I see now. This takes care of decryption and everything :+1:

Reply:
Hmm, thinking about this a bit more, I wonder if we should move _readEventById into `TimelineReader` as a public method? In _readEventById in here we would just create a new ephemeral TimelineReader to read a single event, and in the timeline we have a TimelineReader as well. TimelineReader already takes care of decryption as well.